### PR TITLE
chore: Migrate to ESLint.

### PR DESCRIPTION
**Pull request by darkobits**
This update migrates the repository to [`ts-unified`](https://github.com/darkobits/ts-unified), allowing us to outsource several development dependencies and base configurations for tooling. As part of this update, the project is also moving from TSLint (now deprecated) to ESLint, which recently added support for TypeScript. This greatly increases our ability to maintain quality as the code base changes.

---

#### Summary of Changes

* Install `ts-unified` and update tooling configuration files accordingly.
* Removed development dependencies now provided by `ts-unified`. ✨ 
* Fix any linting errors raised by ESLint.
* Update all dependencies and address any breaking changes.

### CODE REVIEW

**Comment by darkobits** [comment]
@romainricard This PR **should not** change the appearance or behavior of the site in any way. All changes to source files are intended to fix linting errors (re: make the code more readable, enforce best practices). If you see anything in the diff that you suspect will result in a change in behavior or appearance, please don't hesitate to call it out.

File: src/components/contribute/Contribute.tsx [hunk 1]
```diff
@@ -285,9 +289,9 @@ const ContributeComponent: React.FunctionComponent = () => {
               {/* [Field] NSFW */}
               <div className="form-group">
                 <div className="form-row">
-                  <label className={cx('col-12', 'mb-2', errors.isNsfw && 'text-danger')}>
+                  <legend className={cx('col-12', 'mb-2', errors.isNsfw && 'text-danger')}>
```
Comment (darkobits) [inline src/components/contribute/Contribute.tsx:292]: This change may seem a little counter-intuitive. The [`jsx-a11y/label-has-associated-control`](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-associated-control.md) rule requires that, for accessibility reasons, any `<label>` tag have a `for` attribute pointing to an HTML control. When using radio buttons, the `<label>` tags linked to the `<input>` controls are the labels for the options themselves. Therefore, the text for the prompt needs to use a different tag, and the preferred tag for this use case is `<legend>`.

**Comment by romainricard** [comment]
Thanks @darkobits! I added a commit to fix a minor bug: spaces missing before/after line break with a new tag.

```html
            on third-parties, so we can't guarantee that <code>signalstickers.com</code>
            will be accessible at any time.
```

image

If there is another, more elegant fix, do not hesitate to revert my commit.
If it is OK for you, let me know and I'll merge!

**Comment by darkobits** [comment]
Looks good!

**Comment by romainricard** [comment]
Well, :handshake: then.

**Pull request merged**